extensions/amp-form/0.1/amp-form.js

```diff
@@ -1223,6 +1229,80 @@ export class AmpForm {
     }
   }
 
+  /**
+   * Initialize form fields from query parameter values if attribute
+   * 'data-initialize-from-url' is present on the form and attribute
+   * 'data-allow-initialization' is present on the field.
+   * @private
+   */
+  maybeInitializeFromUrl_() {
+    if (
+      isProxyOrigin(this.win_.location) ||
+      !this.form_.hasAttribute('data-initialize-from-url')
+    ) {
+      return;
+    }
+
+    const valueTags = ['SELECT', 'TEXTAREA'];
+    const valueInputTypes = [
+      'color',
+      'date',
+      'datetime-local',
+      'email',
+      'hidden',
+      'month',
+      'number',
+      'range',
+      'search',
+      'tel',
+      'text',
+      'time',
+      'url',
+      'week',
+    ];
+    const checkedInputTypes = ['checkbox', 'radio'];
```

Comment:
This is discussed in https://github.com/ampproject/amphtml/issues/24876#issuecomment-547320066 . Since `:valid` and even `[value]` (for hidden input) selectors are just as powerful as `:checked`, there's not enough reason to exclude just checkbox style inputs. I've included examples of these selectors in a [pr-deploy-bot sample](https://storage.googleapis.com/amp-test-website-1/amp_dist_66181/examples/amp-form-initialization-test.html?input-color=%23ffffff&input-datetime-local=2019-01-01T00%3A00&input-email=test%40test.com&input-month=2019-01&input-number=100&input-range=6&input-search=some+search&input-tel=%2B1-555-555-5555&input-text=some+text&input-time=01%3A00&input-url=https%3A%2F%2Famp.dev%2Fs%3Fq%3D3%23e&input-week=2019-W01&input-hidden=some+hidden&input-checkbox-1=1&input-checkbox-2=2&input-radio=2&select=2&textarea=some+textarea&nofill-input-text=other+text&nofill-input-checkbox-1=1&nofill-input-checkbox-2=2&nofill-input-radio=2&nofill-select=2&nofill-textarea=other+textarea&layout-checked=on&layout-invalid=.&layout-hidden=0).

Reply:
Sorry, missed that thread! Thanks for pointing me to it.

Reply:
I think we can change the property value without changing the attribute value, which avoids triggering attribute selectors. E.g.

```js
checkbox.checked = true;
checkbox.matches('[checked]'); // false
```

`:valid` might still be an issue but it's more of an edge case IMO.